src/Castle.Core/DynamicProxy/Generators/MethodWithInvocationGenerator.cs

```diff
@@ -32,16 +32,26 @@ public class MethodWithInvocationGenerator : MethodGenerator
 	{
 		private readonly IInvocationCreationContributor contributor;
 		private readonly GetTargetExpressionDelegate getTargetExpression;
+		private readonly GetTargetExpressionDelegate getTargetTypeExpression;
 		private readonly Reference interceptors;
 		private readonly Type invocation;
 
 		public MethodWithInvocationGenerator(MetaMethod method, Reference interceptors, Type invocation,
 		                                     GetTargetExpressionDelegate getTargetExpression,
 		                                     OverrideMethodDelegate createMethod, IInvocationCreationContributor contributor)
+			: this(method, interceptors, invocation, getTargetExpression, null, createMethod, contributor)
+		{
+		}
```

Comment:
Adding the new parameter to the existing (publicly visible) ctor would be a breaking change, hence a new ctor.

Reply:
No one should be using these classes anyway as they really are internal implementation, but happy for this to stay as is for now and can be removed later.